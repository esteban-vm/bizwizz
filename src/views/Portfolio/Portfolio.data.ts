import type { WorkProps } from './Work'

const works: WorkProps[] = [
  {
    id: '56d05460-059d-4ea1-b980-8c9035d05aa2',
    image: { src: '/images/portfolio-image-1.jpg', height: 3_078, width: 4_309 },
  },
  {
    id: 'd2066021-f1f0-4963-8832-177977a0e99c',
    image: { src: '/images/portfolio-image-2.jpg', height: 4_000, width: 6_000 },
  },
  {
    id: 'cfe97174-6585-45e4-8161-c19161428611',
    image: { src: '/images/portfolio-image-3.jpg', height: 6_720, width: 4_480 },
  },
  {
    id: '7fd58301-7a11-481b-8b3b-46e61fbc25e2',
    image: { src: '/images/portfolio-image-4.jpg', height: 3_072, width: 4_608 },
  },
  {
    id: 'b86757c5-14ab-4479-96f8-063cbfd6d16b',
    image: { src: '/images/portfolio-image-5.jpg', height: 5_255, width: 3_792 },
  },
  {
    id: '81bafebd-69da-47af-88c5-26d4bd400dbf',
    image: { src: '/images/portfolio-image-6.jpg', height: 3_000, width: 2_000 },
  },
  {
    id: 'fc1e2205-854c-4a5e-b100-eadfb262ea83',
    image: { src: '/images/portfolio-image-7.jpg', height: 8_320, width: 6_240 },
  },
  {
    id: 'dc7a5fc1-e1ce-4ddc-a862-512b20846d2b',
    image: { src: '/images/portfolio-image-8.jpg', height: 6_000, width: 4_000 },
  },
]

export default works
