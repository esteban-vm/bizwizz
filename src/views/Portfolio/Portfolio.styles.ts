import clsx from 'clsx'

const styles = {
  works: clsx('flex items-center justify-center gap-3'),
  work_column: clsx('basis-1/5'),
}

export default styles
