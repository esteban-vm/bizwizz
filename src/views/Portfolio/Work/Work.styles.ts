import clsx from 'clsx'

const styles = {
  wrapper: clsx('m-3 w-full overflow-hidden'),
  image: clsx(
    'h-auto w-full',
    'brightness-100 contrast-100 saturate-0 transition-all duration-500',
    'hover:scale-125 hover:filter-none'
  ),
}

export default styles
