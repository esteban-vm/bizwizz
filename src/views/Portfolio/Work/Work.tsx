import type { FC, ID, ImageData } from '@/types'
import Image from 'next/image'
import styles from './Work.styles'

type WorkImage = `portfolio-image-${1 | 2 | 3 | 4 | 5 | 6 | 7 | 8}`

export type WorkProps = {
  id: ID
  image: ImageData<WorkImage>
}

const Work: FC<WorkProps> = ({ id, image }) => {
  return (
    <article className={styles.wrapper} id={id}>
      <Image alt='' className={styles.image} {...image} />
    </article>
  )
}

export default Work
