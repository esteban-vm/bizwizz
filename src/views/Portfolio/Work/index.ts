export { default, type WorkProps } from './Work'
