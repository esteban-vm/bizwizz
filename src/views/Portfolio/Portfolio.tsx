import type { FC } from '@/types'
import { Wave } from '@/components'
import Work from './Work'
import works from './Portfolio.data'
import styles from './Portfolio.styles'

const Portfolio: FC = () => {
  return (
    <section aria-labelledby='portfolio_title' className={styles.wrapper('white')} id='portfolio'>
      <Wave is='black' />
      <div className={styles.header}>
        <h2 className={styles.heading} id='portfolio_title'>
          Our popular projects
        </h2>
        <h3 className={styles.subheading}>Lorem ipsum dolor sit amet</h3>
      </div>
      <div className={styles.works}>
        <div className={styles.work_column}>
          {works.slice(undefined, 2).map((work) => (
            <Work key={work.id} {...work} />
          ))}
        </div>
        <div className={styles.work_column}>
          {works.slice(2, 4).map((work) => (
            <Work key={work.id} {...work} />
          ))}
        </div>
        <div className={styles.work_column}>
          {works.slice(4, 6).map((work) => (
            <Work key={work.id} {...work} />
          ))}
        </div>
        <div className={styles.work_column}>
          {works.slice(6).map((work) => (
            <Work key={work.id} {...work} />
          ))}
        </div>
      </div>
    </section>
  )
}

export default Portfolio
