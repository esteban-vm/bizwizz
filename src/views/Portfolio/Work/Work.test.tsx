import { render, cleanup, within, screen } from '@/tests'
import { works } from '../Portfolio'
import Work from './Work'

describe('Work:', () => {
  let asFragment: () => DocumentFragment
  const [testWork] = works

  beforeEach(() => {
    void ({ asFragment } = render(<Work {...testWork} />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const teamMateElem = screen.getByRole('article')
    expect(teamMateElem).toBeInTheDocument()
    expect(teamMateElem).toHaveAttribute('id', testWork.id)
    expect(within(teamMateElem).getByRole('img')).toBeInTheDocument()
  })
})
