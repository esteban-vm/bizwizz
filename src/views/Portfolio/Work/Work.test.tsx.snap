// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Work: > should render correctly 1`] = `
<DocumentFragment>
  <article
    class="m-3 w-full overflow-hidden"
    id="56d05460-059d-4ea1-b980-8c9035d05aa2"
  >
    <img
      alt=""
      class="h-auto w-full brightness-100 contrast-100 saturate-0 transition-all duration-500 hover:scale-125 hover:filter-none"
      data-nimg="1"
      decoding="async"
      height="3078"
      loading="lazy"
      src="/_next/image?url=%2Fimages%2Fportfolio-image-1.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fimages%2Fportfolio-image-1.jpg&w=3840&q=75 1x"
      style="color: transparent;"
      width="4309"
    />
  </article>
</DocumentFragment>
`;
